data/examples/declaration/value/function/lambda-multi-line-out.hs

```diff
@@ -23,5 +23,5 @@ tricky2 =
 
 foo =
   prop "is inverse to closure" $ \(f :: StaticPtr (Int -> Int))
-  (x :: Int) ->
+                                  (x :: Int) ->
```

Comment:
It looks like this one is against "minimal diffs" strategy. Imagine that the prefix part

```
prop "is inverse to closure"
```

is changed so that its length has changed. In that case all lambda arguments will be shifted, i.e. we have a case of diff amplification. I'd imagine that we also should insert a newline after `$` and before `\(...)` part so they go on the next line and one indentation level to the right.

Reply:
That is a good point. I changed the logic to not hang lambdas if the argument list spans multiple lines. 

However it behaves slightly different than how you mentioned; the newline is before the `$` rather than after. Now we get:

```haskell
foo =
  prop "is inverse to closure"
    $ \(f :: StaticPtr (Int -> Int))
       (x :: Int) ->
        bar
```

I think this behaviour makes more sense since we usally place operators at the beginning of the line.

What do you think?